src/librustdoc/clean/mod.rs

```diff
@@ -169,11 +169,19 @@ impl Clean<ExternalCrate> for CrateNum {
                 for attr in attrs.lists(sym::doc) {
                     if let Some(v) = attr.value_str() {
                         if attr.has_name(sym::keyword) {
-                            if v.is_doc_keyword() {
-                                keyword = Some(v.to_string());
-                                break;
+                            let k = v.to_string();
+                            if !rustc_lexer::is_ident(&k) {
+                                cx.tcx
+                                    .sess
+                                    .struct_span_err(
+                                        attr.span(),
+                                        &format!("`{}` is not a valid ident", v),
```

Comment:
This is how that might be achieved (there might be a better way; I don't know the ast well enough. Outlining that into its own `span_of_value_str` might make it more readable):

```suggestion
                                        attr.meta_item().unwrap().name_value_literal().unwrap().span,
                                        &format!("`{}` is not a valid identifier", v),
```

Reply:
Funnily enough, we generally don't indicate the value but the attribute. Might be interesting to improve it in some other places. Noting that for myself haha.